sway-core/src/language/parsed/expression/mod.rs

```diff
@@ -29,6 +29,201 @@ pub struct Expression {
     pub span: Span,
 }
 
+impl Expression {
```

Comment:
I assume this was used before in the previous approach when the AST was built manually in code. Should we leave it or remove it? That's a lot of code that could be used for desugaring in the future or refactoring existing desugaring. Perhaps moved to a dedicated module, turned into builder pattern, ... I don't know. I am afraid if it just stays where it is now and unused it will remain dead code.

Reply:
Removed.